hazelcast/src/test/java/com/hazelcast/test/AfterClassesStatement.java

```diff
@@ -44,6 +45,7 @@ public void evaluate() throws Throwable {
         if (!instances.isEmpty()) {
             String message = "Instances haven't been shut down: " + instances;
             HazelcastInstanceFactory.terminateAll();
+            assertTrueEventually(HazelcastTestSupport::assertNoRunningInstances, 30);
```

Comment:
:+1: 

We could do the same for the clients below.

Reply:
Added: https://github.com/hazelcast/hazelcast/pull/23043/commits/784628f8b217c9e55a0f96c5650ea41b2a878782